prestoadmin/server.py

```diff
@@ -82,11 +85,218 @@ def new_sysnode_processor(node_info_rows):
 CONNECTOR_INFO_SQL = 'select catalog_name from system.metadata.catalogs'
 _LOGGER = logging.getLogger(__name__)
 
+DOWNLOAD_DIRECTORY = main_dir
+DEFAULT_RPM_NAME = 'presto-server-rpm.rpm'
+
+
+def _find_local_presto_rpm(local_path):
+    if os.path.isfile(local_path):
+        return local_path
+
+    path_relative_to_download_dir = os.path.join(DOWNLOAD_DIRECTORY, local_path)
+    if os.path.isfile(path_relative_to_download_dir):
+        return path_relative_to_download_dir
+
+    return None
+
+
+def _find_or_download_latest_presto_rpm(rpm_source):
+    if rpm_source == 'facebook':
+        return _find_or_download_latest_presto_rpm_facebook()
+    elif rpm_source == 'teradata':
+        return _find_or_download_latest_presto_rpm_teradata()
+    else:
+        return None
+
+
+def _find_or_download_latest_presto_rpm_facebook():
+    latest_rpm_url = 'https://repository.sonatype.org/service/local/artifact/maven' \
+                     '/content?r=central-proxy&g=com.facebook.presto' \
+                     '&a=presto-server-rpm&e=rpm&v=RELEASE'
+    return _find_or_download_rpm(latest_rpm_url)
+
+
+def _find_or_download_latest_presto_rpm_teradata():
+    abort('Finding and downloading latest teradata presto rpm is currently not supported.\n'
+          'Try again after downloading the rpm and specifying the local path to the rpm')
+
+
+def _check_good_response_status(response_status):
+    return response_status == 200
```

Comment:
How about adding a warn() with the specific URL we tried to aid troubleshooting? Or log it only. Either seems OK.